### PR TITLE
store: make DeleteContainer parallel

### PR DESCRIPTION
**Pull request by giuseppe**
the operations have no dependency on each other so they can be
performed in parallel.

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@saschagrunert @nalind @vrothberg PTAL

**Comment by giuseppe** [comment]
marked as WIP for now, need to think more of it.

The cost I want to avoid is the double sync that each store does on the atomic write

File: store.go [hunk 1]
```diff
@@ -2320,24 +2320,53 @@ func (s *store) DeleteContainer(id string) error {
 
 	if rcstore.Exists(id) {
 		if container, err := rcstore.Get(id); err == nil {
+			var (
+				wg        sync.WaitGroup
+				rlError   error
+				rcError   error
+				rmGcError error
+				rmRcError error
+			)
```
Comment (vrothberg) [inline store.go]: Can we replace the variables above with one error channel? I think that would be a bit more idiomatic and ultimately less lines of code. We can wrap the errors into one before returning.

Comment (giuseppe) [reply store.go]: yes sure, fixed in the new version.

**Comment by rhatdan** [comment]
@saschagrunert PTAL

**Comment by nalind** [comment]
It gets harder to clean up things if we fail to delete the layer for any reason.  I don't know how often that comes up, but it leaves me a bit uneasy.

I'm not entirely sure about doing the waitgroup `Wait` in a goroutine and using a channel instead of just using variables to hold the nil-or-error return values we want to examine, but that's not a blocker.

**Comment by saschagrunert** [comment]
> I'm not entirely sure about doing the waitgroup `Wait` in a goroutine and using a channel instead of just using variables to hold the nil-or-error return values we want to examine, but that's not a blocker.

I thought the same and concluded for myself that a wait group is appropriate here. :innocent:

**Comment by rhatdan** [comment]
@nalind Did you just give a luke warm LGTM?

**Comment by nalind** [comment]
The inability to retry a failed cleanup still makes me uneasy, but I'm not going to block on that.

**Pull request merged**